raiden/ui/cli.py

```diff
@@ -1025,8 +1035,21 @@ def _run_app(self):
                 )
         finally:
             self._shutdown_hook()
-            if api_server:
-                api_server.stop()
+
+            def stop_task(task):
+                try:
+                    if isinstance(task, Runnable):
+                        task.stop()
+                    else:
+                        task.kill()
+                finally:
+                    task.get()  # re-raise
```

Comment:
That's intended, `GreenletExit` isn't properly an exception, and not re-raising it is fine.

Reply:
How is it not a proper exception?

Reply:
It's a way for `gevent` to tell greenlets they were killed, and aren't intended to be re-raised (even because `get` doesn't re-raise them). It's a way for `gevent` to communicate with code running inside a task, and is handled differently from _proper_ exceptions inside it.